addons/web/static/src/js/views/kanban/kanban_model.js

```diff
@@ -292,6 +292,13 @@ var KanbanModel = BasicModel.extend({
         if (element.parentID) {
             element.limit = this.loadParams.limit;
         }
+        // if list datapoint is reloaded then remove activeFilter from all child datapoint
+        if (element.type === 'list' && !element.parentID && !options || options && !('activeFilter' in options)) {
+            _.each(element.data, (elemId) => {
+                const elem = this.localData[elemId];
+                elem.activeFilter = { domain: [], value: false };
+            });
+        }
```

Comment:
```suggestion
```

The real problem here is that the kanban column does not keep its previous activeFilter. I don't think this information is lost, I just think that it is not restored.

Try keeping the localData activeFilter record by adding the following somewhere in kanban_modal __get method:
```js
// Always keep activeFilter
result.activeFilter = dp.activeFilter;
```

Then in the init of kanban_column_progressbar, try that:
```diff
- this.activeFilter = state.activeFilter;
+ this.activeFilter = columnState.activeFilter || state.activeFilter;
```

Then you should have a good starting point.